### DISCUSSION TITLE
Error : ORA-01732: data manipulation operation not legal on this view

### DISCUSSION THREAD

**Discussion by HakanKaraoglu** [nhibernate/nhibernate-core#2828] [Q&A]
I am running a query like below, but when I run this query with different values for the second time, I get an error.

I noticed something like this, the first query works successfully, I even see the select query, but when I run it for the second time, it goes as an update. Although the second query is a select query, it is an interesting situation, frankly I can't understand it.

Method:

```
public override async Task<ConcurrentDictionary<int, ConcurrentBag<FeedKeywordProxy>>> GetWorkspaceAsync(FeedProxy feedProxy, string[] tags)
        {
            if (tags != null && tags.Length > 0)
            {
                var feedKeywordList = new ConcurrentBag<FeedKeywordProxy>();

                for (int j = 0; j < tags.Length; j++)
                {
                    var tag = tags[j];
                    var keywords = await _nhUnitOfWork.KeywordDetectionViewRepository.GetWorkspacesByTag(feedProxy.Source, tag, feedProxy.Language);
                    for (int i = 0; i < keywords.Count; i++)
                    {
                        var workspaceKeyword = keywords[i];
                        var feedKeyword = CreateFeedKeywordInstance(workspaceKeyword);
                        feedKeywordList.Add(feedKeyword);
                    }
                }
                return GroupFeedKeywordsByWorkspaceId(feedKeywordList);
            }
            return new ConcurrentDictionary<int, ConcurrentBag<FeedKeywordProxy>>();
        }

```

Query:

```
 public async Task<IList<KeywordDetectionView>> GetWorkspacesByTag(string source, string tag, LanguageTypes languageTypes)
        {
            var query = _session.CreateCriteria<KeywordDetectionView>();

            int keywordId = int.Parse(tag.Substring(2));

            if (tag.StartsWith("k_"))
            {
                query.Add(Restrictions.Eq("KeywordId", keywordId));
            }
            else if (tag.StartsWith("v_"))
            {
                query.Add(Restrictions.Eq("VariationId", keywordId));
            }
            query.Add(Restrictions.Eq("SourceCode", source));
            query.Add(Restrictions.Or(Restrictions.Eq("LanguageCode", languageTypes.ToString().ToLowerInvariant()), Restrictions.IsNull("LanguageCode")));

            return await query.SetCacheable(false).ListAsync<KeywordDetectionView>();
        }
```

1. Loop Profiler Result :

SELECT this_.ID as id1_20_0_, this_.WORKSPACEID as workspaceid2_20_0_, this_.KEYWORDID as keywordid3_20_0_, this_.VARIATIONID as variationid4_20_0_, this_.SOURCEID as sourceid5_20_0_, this_.SOURCECODE as sourcecode6_20_0_, this_.LANGUAGEID as languageid7_20_0_, this_.LANGUAGECODE as languagecode8_20_0_, this_.KEYWORDTEXT as keywordtext9_20_0_, this_.VARIATIONTEXT as variationtext10_20_0_ FROM VIEWNAME this_ WHERE this_.KEYWORDID = ? and this_.SOURCECODE = ? and (this_.LANGUAGECODE = ? or this_.LANGUAGECODE is null)

2. Loop Profiler Result and Error:

UPDATE VIEWNAME SET WORKSPACEID = ?, KEYWORDID = ?, VARIATIONID = ?, SOURCEID = ?, SOURCECODE = ?, LANGUAGEID = ?, LANGUAGECODE = ?, KEYWORDTEXT = ?, VARIATIONTEXT = ? WHERE ID = ?

ORA-01732: data manipulation operation not legal on this view, 

But i am not sending update query

**Comment by bahusoid** [comment]
> But i am not sending update query

It might be triggered by auto flush. Most likely you have nullable columns that's are mapped to not nullable properties. It might make NHIbernate believe that your entity is dirty (for instance original loaded value is null but in entity it's stored as 0). But as your view is not updateable just set  `mutable="false"` in mapping.
